### DISCUSSION TITLE
How to change the label of an Entity?

### DISCUSSION THREAD

**Discussion by robertoshimizu** [explosion/spaCy#5176] [Help: Coding & Implementations]
I've came across this example:
```import spacy
from spacy.tokens import Span

nlp = spacy.load("en_core_web_sm")
doc = nlp("Tesla CEO Elon Musk said Wednesday on Twitter that\
           the company would make ventilators in case of a shortage\
           during the coronavirus pandemic. Now, New York City Mayor\
           Bill de Blasio is holding his feet to the fire.")
```
and it resulted in:

--------------  ---  ---  --------
Tesla             0    5  CARDINAL
Elon Musk        10   19  PERSON
Wednesday        25   34  DATE
New York City   166  179  GPE
Bill de Blasio  196  210  PERSON
--------------  ---  ---  --------
I tried to change the label for Tesla from CARDINAL to ORG. For this I tried: 1) create a new Span, however when adding to doc.ent, however it conflicts with the former one. I also tried to remove it but it seems that doc.ents is an object. Any suggestions on how do I make this change?

## Which page or section is this issue related to?
https://spacy.io/usage/linguistic-features#named-entities

**Comment by svlandeg** [comment]
You're right that you can't easily change the `label_` of a `Span` because spans are views, and you can't easily change `doc.ents` because it's a tuple.

The easiest work-around is by converting the `ents` to a list, make the changes, and store them by setting `doc.ents` again.
```
ents = list(doc.ents)
old_ent = ents[0]
new_ent = Span(doc, old_ent.start, old_ent.end, label="ORG")
ents[0] = new_ent
doc.ents = ents
```
